tests/test_util_functions.py

```diff
@@ -0,0 +1,119 @@
+    """
+    saved_state = {
+        'third_party_auth': None,
+        'third_party_auth.models': None,
+    }
+
+    def __init__(self, providers):
+        """
+        initialize list of providers, this is the list queryset will return.
+        """
+        self.providers = providers
+
+    def __enter__(self):
+        """
+        Save existing package reference, and mock third_party_auth package.
+        """
+        self.saved_state['third_party_auth'] = sys.modules.get('third_party_auth')
+        self.saved_state['third_party_auth.models'] = sys.modules.get('third_party_auth.models')
+
+        sys.modules['third_party_auth'] = mock.Mock()
+        sys.modules['third_party_auth.models'] = mock.Mock(
+            SAMLProviderConfig=mock_saml_provider_config(self.providers),
+        )
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        """
+        Restore third_party_auth package references.
+        """
+        sys.modules['third_party_auth'] = self.saved_state['third_party_auth']
+        sys.modules['third_party_auth.models'] = self.saved_state['third_party_auth.models']
+
+
+def mock_saml_provider_config(providers):
+    """
+    get mock class for SAMLProviderConfig model.
+    """
+    class SAMLProviderConfig(object):
+        """
+        mock class for SAMLProviderConfig.
+        """
+        objects = mock.Mock(
+            current_set=mock.Mock(return_value=mock.Mock(
+                filter=mock.Mock(
+                    return_value=mock.Mock(all=mock.Mock(
+                        return_value=providers
+                    ))
+                )
+            ))
+        )
+    return SAMLProviderConfig
+
+
+class TestUtils(unittest.TestCase):
+    """
+    Tests for utility functions.
+    """
+    def test_get_idp_choices(self):
+        """
+        Test get_idp_choices returns correct options for choice field or returns None if
+        thirdParty_auth is not installed.
+        """
+        options = utils.get_idp_choices()
+        self.assertIsNone(options)
+        expected_list = [('', '-'*7), ('test1', 'test1'), ('test2', 'test2')]
+
+        with mock.patch('enterprise.utils.get_available_idps', mock_get_available_idps(['test1', 'test2'])):
+            choices = utils.get_idp_choices()
+            self.assertListEqual(choices, expected_list)
+
+        available_providers = mock_get_available_idps(['test1', 'test2'])()
+
+        with MockThirdPartyAuth(available_providers):
+            self.assertListEqual(expected_list, utils.get_idp_choices())
+
+    def test_get_available_idps(self):
+        """
+        Test get_available_idps returns correct value or raises a ValueError if
+        thirdParty_auth is not installed.
+        """
+        with self.assertRaises(ValueError):
+            utils.get_available_idps()
+
+        expected_list = ['test1', 'test2']
+
+        with MockThirdPartyAuth(expected_list):
+            self.assertListEqual(expected_list, utils.get_available_idps())
```

Comment:
@saleem-latif nit: swap `expected_list` and `utils.get_available_idps()`. I believe first parameter should be actual value and second parameter - expected value.

Also, it might make sense to start using [pytest assertions](http://doc.pytest.org/en/latest/assert.html) (`assert somerthing`) instead of unittest assertions (`self.assertSomething`)